be/src/vec/data_types/data_type_number_base.h

```diff
@@ -188,12 +189,12 @@ class DataTypeNumberBase : public IDataType {
         for (int row_num = 0; row_num < size; row_num++) {
             auto num = is_const ? col_vec.get_element(0) : col_vec.get_element(row_num);
             static_cast<const Derived*>(this)->push_number(chars, num);
-            offsets[row_num] = chars.size();
+            offsets[row_num] = cast_set<UInt32>(chars.size());
```

Comment:
seems we only need static_cast? push_number can check the chars is over uint32

Reply:
done